flax/linen/sequential.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2021 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence
+from flax.linen.module import Module
+from typing import Type
+
+
+class Sequential(Module):
+
+  """ A sequential module for stacking layers sequentially. Each layer has only one input and one output. Layers will be added to it in the order they are passed in.
+    Attributes:
+      layers: A sequence of the layers that needed to be stacked sequentially.
+  """
+  layers: Sequence[Type[Module]]
```

Comment:
It seems `layers` is a sequence of Module instances, so shouldn't this just be `Sequence[Module]`?

Reply:
`Sequence[Module]` fails in building due to a pytype error. as 'Module' object is not callable. but it works well if you pass in Module instances. so, `Sequence[Type[Module]]` is used to pass instances of Module object.

Reply:
If you look at the documentation for typing.Type:
https://docs.python.org/3/library/typing.html#typing.Type
you'll see that it is used to represent class objects, not instances.

If you set layers as
```
  layers: Sequence[Type[Module]]
```

I think the issue is that you are doing `layer(x)`, which assumes that `__call__` is defined as a method.
While this is common for most modules (like Dense), the base class `Module` does not define `__call__` and you can define modules which do not have a `_call__` method.

When you declare layers as `Sequence[Type[Module]]` you are tricking pytype to think `layer(x)` is OK the same way that `nn.Dense(3)` is OK -- i.e. because classes define constructors.

Reply:
Is there any reason `layers` has to be a `Sequence[Module]` instead of `Sequence[Callable]`? For example, users could use `lambda x: flax.linen.max_pool(x, (2, 2))` as a layer as well.

Reply:
mark this as Sequence[Callable] - that should work now w. typechecking and Modules